lib/pure/options.nim

```diff
@@ -67,6 +67,34 @@
 ##     assert(false)  # This will not be reached
 ##   except UnpackError:  # Because an exception is raised
 ##     discard
+##
+## When you have a value of type ``Option[T]`` and you want to have
+## it's value of type ``T``, you can access the value with ``get``.
+## If you don't provide a fallback value there will be an exception.
+## With the ``or`` operator you can chain together several expressions
+## of type ``Option[T]`` and the whole expression will evaluate the
+## operand from left to right until an the operand that is not
+## none.  This operand will be returned.  If all operands are none, then
+## the expression evaluates to none. For convenience the last operand
+## of such an expression does not need to be of type ``Option[T]``, it
+## can be just of type ``T`` for the last fallback. If that is the
+## case the whole expression will be of type ``T``, because it could
+## not evaluate to none anymore.
```

Comment:
Your docs, while appreciated, are far too verbose. 

Create a short, and to the point heading:

```nim
## Retrieving values from an option
## ========================
##
## Values can be retrieved in many ways, primarily using ``get`` or the ``or`` operator.
##
## Using ``get``
## --------------
##
## This is the simplest approach:
##
## ..code-block:: ....
##
##
## But it means there is an exception raised when the option isn't a ``some``.
##
## Using ``or``
## -------------
##
## ....

Reply:
Short sentences, and short specific examples are always better than a huge paragraph and a huge example.